### PR TITLE
feat(vue-model-api): add a library to use models from Vue.js

### PR DESCRIPTION
**Pull request by odzhychko**
Users can use the exported functions `useModelsFromJson` to read models from exported JSON files.

To connect to a server, users must use `useModelClient` and `useRootNode`.

The returned objects also work with Vue.js reactivity system and the typed classes generated by
`model-api-gen`.

### CODE REVIEW

File: vue-model-api/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+
+plugins {
```
Comment (odzhychko) [inline vue-model-api/build.gradle.kts:4]: Look at task dependencies.
Declare inputs and ouputs.

File: vue-model-api/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    alias(libs.plugins.node)
+    alias(libs.plugins.npm.publish)
+}
+
+tasks.npmInstall {
+    dependsOn(":ts-model-api:build")
+    dependsOn(":model-client:packJsPackage")
+}
+
+tasks.named("npm_run_build") {
```
Comment (odzhychko) [inline vue-model-api/build.gradle.kts:15]: Add test for interopt with generated API.

File: vue-model-api/src/useRootNode.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * @param repositoryId - Reactive reference of a repositoryId on the model server.
+ * @param branchId - Reactive reference of a branchId in the repository of the model server.
+ *
+ * @returns {Object} values Wrapper around diffrent returned values.
+ * @returns {Ref<ClientJS>} values.rootNode  Reactive reference to a reactive root node.
+ * @returns {() => void} values.dispose A function to manually dispose the root node.
+ * @returns {Ref<unknown>} values.error Reactive reference to a connection error.
+ */
+export function useRootNode(
+  client: MaybeRefOrGetter<ClientJS | null>,
```
Comment (odzhychko) [inline vue-model-api/src/useRootNode.ts:33]: Test this shit.

**Comment by odzhychko** [comment]
Rewrite test to follow Arrange-Act-Assert
Enable caching for all tasks

**Comment by odzhychko** [comment]
do not run prettier in check, it is already run by pre commit checks

File: model-api-gen-gradle-test/vue-integration/src/intergration.test.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            name: "aName",
+          },
+        },
+      ],
+    },
+  };
+
+  const untypedNode = useRootNode(nodeData).getChildren("children1")[0];
+  const typedNode = LanguageRegistry.INSTANCE.wrapNode(untypedNode);
+  if (!(typedNode instanceof _N_TypedImpl_INamedConceptImpl)) {
```
Comment (odzhychko) [inline model-api-gen-gradle-test/vue-integration/src/intergration.test.ts:40]: Replace with type guards

**Comment by odzhychko** [comment]
Mark as @alpha instead of @experimental

**Comment by slisson** [comment]
The dependency tracking of read/write access to nodes is something we need also when not using vue.js. I implemented something similar for the text-editor that also integrates with [modelix.incremental](https://github.com/modelix/modelix.incremental):

https://github.com/modelix/modelix.text-editor/blob/main/projectional-editor/src/commonMain/kotlin/org/modelix/editor/IncrementalBranch.kt

https://github.com/modelix/modelix.core/pull/273/files

We should reuse that for the vue.js integration.

**Review by slisson — DISMISSED** [review]

**Review by slisson — APPROVED** [review]

**Pull request merged**